### PR TITLE
Make function names more consistent 

### PR DESCRIPTION
**Pull request by FirasSahliADLinktech**
To make function names more more consistent the following functions were renamed:
* dds_topic_get_filter to dds_get_topic_filter and dds_topic_set_filter to dds_set_topic_filter, * dds_get_enabled_status to dds_get_status_mask, dds_set_enabled_status to dds_set_status_mask
* and dds_instance_lookup to dds_lookup_instance

### CODE REVIEW

**Comment by k0ekk0ek** [comment]
@FirasSahliADLinktech thanks for your pull request. However, please add a proxy and a deprecation warning when someone uses the old interface. Apart from that please explain why these changes should be applied. Thanks.

File: src/core/ddsc/tests/entity_status.c [hunk 1]
```diff
@@ -354,32 +354,32 @@ Test(ddsc_entity, inconsistent_topic)
     cr_assert_gt(top, 0, "fails %d", dds_err_nr(top));
```
[the remark this answers is not in the document]
Comment (FirasSahliADLinktech) [reply src/core/ddsc/tests/entity_status.c]: I think this code block should be removed it it is a dead code block

**Comment by k0ekk0ek** [comment]
Also, @FirasSahliADLinktech: integrate the changes into the other commits. I don't like first introducing errors into the repository and then in the same pull request have commit that fixes them. Thanks.

**Comment by k0ekk0ek** [comment]
I'm ok with the changes if the AppVeyor builds pass (without newly introduced warnings).

**Comment by FirasSahliADLinktech** [comment]
@eboasson  I am still waiting for this pull request to be merged. Can you please review it and merge it if it is good.

**Comment by eboasson** [comment]
@FirasSahliADLinktech, apologies ... it slipped my mind.

**Comment by k0ekk0ek** [comment]
> I'm ok with the changes if the AppVeyor builds pass (without newly introduced warnings).

@FirasSahliADLinktech. The change introduced rather a lot of warnings. Please be a little more careful next time. As for the source of the warnings, I have a fix included in a pull request I'll submit today.

**Comment by eboasson** [comment]
@k0ekk0ek thanks for catching that ... that'll teach to me to always also double-check a Windows build whenever something *seems* ok and works fine on Linux/macOS. Unless you can find a way to make Windows builds treat warnings as errors. In that case, the CI builds will save me the checking :)

**Pull request merged**